src/components/Header/Header.tsx

```diff
@@ -178,4 +74,106 @@ function Header() {
     </>
 }
 
+function ShowMenuIcon(props: any) {
+    return (
+        <div className="heading-appbar-menu-icon-container">
+        {props.open ? (
+            <IconButton onClick={props.handleDrawerClose}>
+                      {props.theme.direction === 'ltr' ? <ChevronLeftIcon style={{color: 'white', marginTop: '10px', marginLeft: '10px'}}/> : <ChevronRightIcon />}
+                    </IconButton>
+        ) : (
+            <Toolbar  sx={{ paddingRight: '0px' }}>
+            <IconButton
+                color="inherit"
+                aria-label="open drawer"
+                onClick={props.handleDrawerOpen}
+                edge="start"
+                sx={{ mr: 2 }}>
+                <MenuIcon/>
+            </IconButton>
+        </Toolbar>
+        )}
+    </div>
+    )
+}
+
+function TitleAndButtons(props: any) {
+    return (
+        <div className="header-center-screen-container">
+        <div className="header-center-screen-title"
+            onClick={props.navToHome}>
+                DND FORUM
+        </div>
+        {props.width > 400 ? (
+        <>
+        <div className="header-center-screen-wiki"
+            >
+            <button className='wiki-button'
+                onClick={props.handleClickWiki}
+                onMouseEnter={props.handleMouseEnter}
+                onMouseLeave={props.handleMouseLeave}
+            >
+                Wiki
+            </button>
+        </div>
+            <div className="header-center-screen-create">
+                <button className='create-button'
+                    onClick={props.handleClickCreate}
+                    onMouseEnter={props.handleMouseEnter}
+                    onMouseLeave={props.handleMouseLeave}
+                >
+                    Create
+                </button>
+            </div>
+        </>
+        ): (
+            <div/>
+        )}
+
+    </div>
+    )
+} 
+
+function LoginLogoutDisplayName(props: any) {
```

Comment:
The function name for this is a little confusing. What if we broke these down even smaller?